src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -10,90 +9,103 @@
 
 #include "CLI11.hpp"
 
-#include "CoreTypes.hpp"
 #include "QirContext.hpp"
-#include "QirTypes.hpp"
-#include "QirRuntimeApi_I.hpp"
 #include "SimFactory.hpp"
-#include "QirRuntime.hpp"
 
 using namespace Microsoft::Quantum;
 using namespace std;
 
+struct InteropArray
+{
+    int64_t Size;
+    void* Data;
+};
```

Comment:
Consider adding 
either a constructor taking 2 parameters (and initializing both member variables with those parameters), see example below,
or a default constructor initializing both, e.g to 0.
Otherwise there is a risk of creating an instance of `InteropArray` with the memeber variables **un**initialized.
```c++
struct InteropArray
{
    int64_t Size;
    void* Data;
    InteropArray(int64_t size, void* data)
        : Size(size)
        , Data(data)
    {}
};
```
Then the instances can be created like this: 
```c++
unique_ptr<InteropArray> array(new InteropArray(v.size(), v.data()));
```

Reply:
Added constructors to `InteropArray` and to `InteropRange`.